Source/CronosPlayUnreal/Public/DynamicContractActor.h

```diff
@@ -0,0 +1,161 @@
+// Fill out your copyright notice in the Description page of Project Settings.
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "GameFramework/Actor.h"
+#include "DefiWalletCoreActor.h"
+#include "PlayCppSdkLibrary/Include/defi-wallet-core-cpp/src/contract.rs.h"
+#include "PlayCppSdkLibrary/Include/defi-wallet-core-cpp/src/lib.rs.h"
+#include "PlayCppSdkLibrary/Include/defi-wallet-core-cpp/src/nft.rs.h"
+#include "PlayCppSdkLibrary/Include/defi-wallet-core-cpp/src/ethereum.rs.h"
+#include "DynamicContractActor.generated.h"
+
+using namespace std;
+using namespace rust;
+using namespace org::defi_wallet_core;
+
+// dynamic contract send
+DECLARE_DYNAMIC_DELEGATE_TwoParams(FDynamicContractSendDelegate,
+                                   FCronosTransactionReceiptRaw, TxResult,
+                                   FString, Result);
+
+DECLARE_DYNAMIC_DELEGATE_TwoParams(FCallDynamicContractDelegate, FString,
+                                   JsonResult, FString, Result);
+
+UCLASS()
+class CRONOSPLAYUNREAL_API ADynamicContractActor : public AActor {
+    GENERATED_BODY()
+
+  public:
+    /**
+     * @param cronosrpc    cronos rpc server endpoint
+     * @param contractaddress    contract address
+     * @param abijson    contract abi json
+     * @param success whether succeed or not
+     * @param output_message error message, "" if succeed
+     */
+
+    UFUNCTION(BlueprintCallable,
+              meta = (DisplayName = "NewEthContract", Keywords = "Wallet"),
+              Category = "CronosPlayUnreal")
+    void NewEthContract(FString cronosrpc, FString contractaddress,
+                        FString abijson, bool &success,
+                        FString &output_message);
+
+    /**
+     * @param cronosrpc    cronos rpc server endpoint
+     * @param contractaddress    contract address
+     * @param abijson    contract abi json
+     * @param walletindex    wallet index
+     * @param success whether succeed or not
+     * @param output_message error message, "" if succeed
+     */
+    UFUNCTION(BlueprintCallable,
+              meta = (DisplayName = "NewSigningEthContract",
+                      Keywords = "Wallet"),
+              Category = "CronosPlayUnreal")
+    void NewSigningEthContract(FString cronosrpc, FString contractaddress,
+                               FString abijson, int32 walletindex,
+                               bool &success, FString &output_message);
+
+    /**
+     * EncodeDynamicContract for http access
+     * @param rpcserver    cronos rpc server endpoint
+     * @param contactAddress    contract address
+     * @param abiJson    contract abi json
+     * @param functionName    function name
+     * @param functionArgs    function args in json
+         \code
+     * simple json example: Address, String
+     * [{"Address":{"data":"0x00"}},{"Str":{"data":"my"}}]
```

Comment:
for these examples, maybe it'll be also good to write Solidity function signature, so it's more obvious what t hat sample corresponds?

Reply:
ok